packages/shared/lib/core/wallet/utils/isVisibleActivity.ts

```diff
@@ -12,7 +12,10 @@ import {
     TypeFilterOption,
     StatusFilterOption,
     ActivityType,
+    DateFilterOption,
+    DateUnit,
 } from '../enums'
+import { dateIsAfterOtherDate, dateIsBeforeOtherDate, datesOnSameDay } from '@lib/utils/dateUtils'
 
 // Filters activities based on activity properties. If none of the conditionals are valid, then activity is shown.
 export function isVisibleActivity(activity: Activity): boolean {
```

Comment:
This function is too big, can we break it into smaller functions? (for readability enhancement)

Reply:
yes, now the time has come to split it up